### PR TITLE
Add support for federation versions up to 2.6

### PR DESCRIPTION
**Pull request by nogates**
Added support so we can parse fed 2.4, 2.5 and 2.6. schemas... 

Though I haven't added any extra code to mange those new directives or anything like that .

Also, refactored a bit the logic to fallback to the latest avaiable version when using `links` directive, which I think it's better comparing with defaulting to Federation v1

### CODE REVIEW

**Comment by nogates** [comment]
@rafalp Could you have a look when you got a sec? 🙏

File: ariadne/contrib/federation/schema.py [hunk 1]
```diff
@@ -76,25 +76,27 @@ def make_federated_schema(
...
-    elif link and link.group(1) == "https://specs.apollo.dev/federation/v2.3":
-        definitions = load_schema_from_path(
-            os.path.join(dirname, "./definitions/fed2_3.graphql")
+    definitions_folder = os.path.join(dirname, "definitions")
+    definition_files = os.listdir(definitions_folder)
+    definition_files.sort()
+    last_version_file = definition_files[-1]
+
+    if link:
+        # extract version
```
Comment (rafalp) [inline ariadne/contrib/federation/schema.py:85]:
```suggestion
```

File: ariadne/contrib/federation/schema.py [hunk 2]
```diff
@@ -76,25 +76,27 @@ def make_federated_schema(
...
+        fed_version = link.group(1).split("/")[-1]
```
Comment (rafalp) [inline ariadne/contrib/federation/schema.py:86]: Doesn't this change mean that `https://reddit.com/r/something/v2.1` will also pass as Federation v2.1?

Comment (nogates) [reply ariadne/contrib/federation/schema.py:86]: oh yeah! I should validate the link url and that it belongs to the fed spec. good catch! 🙏

**Comment by nogates** [comment]
ready for another review @rafalp. Thanks!

**Review by rafalp — APPROVED** [review]

**Comment by rafalp** [comment]
Thanks!

**Comment by nogates** [comment]
Thanks @rafalp ! May I ask, when are you planning in creating a release? Happy to create a PR to fill in the CHANGELOG if that helps! Thanks!

**Comment by rafalp** [comment]
When its ready™ Goal is to have query execution over GET in next Ariadne release. When that's merged, I'll cut out a release.

**Comment by nogates** [comment]
Awesome, thank you!

**Comment by rafalp** [comment]
It was already part of 0.23 beta 1, but I'll ship final thing today.

**Pull request merged**